cpp/src/Slice/Grammar.y

```diff
@@ -1051,32 +1061,39 @@ class_extends
     ContainerPtr cont = unit->currentContainer();
     TypeList types = cont->lookupType(scoped->v);
     $$ = 0;
-    if(!types.empty())
+    if (!types.empty())
     {
         ClassDeclPtr cl = ClassDeclPtr::dynamicCast(types.front());
-        if(!cl)
+        if (!cl)
         {
-            string msg = "`";
-            msg += scoped->v;
-            msg += "' is not a class";
-            unit->error(msg);
+            if (auto alias = TypeAliasPtr::dynamicCast(types.front()))
```

Comment:
Absolutely! I separated them to keep the variable initialization readable, but I don't really care either way.

Reply:
Nevermind, I spoke too fast and forgot I was initializing a variable in the if statement.
I don't believe C++ allows you to both evaluate an expression and declare a variable in the same if statement... Unless there's some super fancy C++20 syntax I'm unaware of.

Reply:
The feature is called "if statement with initializer" and is available as of C++17:
https://en.cppreference.com/w/cpp/language/if

```
if (auto alias = xxx; alias && !cl)
{
   ... 
}
```

Reply:
Yes, what I meant to say is there's no way to do it AND keep the same order (checking `cl` first). I could change the order, but it seemed wasteful to always perform this cast when 99% of the time, `!cl` is false, and so the cast is pointless to do.
If you really want, I can change it, but the current 2-if version is more efficient and in my opinion also more readable than
`if (auto alias = TypeAliasPtr::dynamicCast(types.front()); alias && !cl)`
So even if it would save us 3 lines of code... I'd personally prefer to leave it as is.

Reply:
For reference, this is the current code in my PR:
```
        if (!cl)
        {
            if (auto alias = TypeAliasPtr::dynamicCast(types.front()))
            {
                ...
            }
        }
```
vs the other way of writing it:
```
        if (auto alias = TypeAliasPtr::dynamicCast(types.front()); alias && !cl)
        {
            ...
        }
```

Reply:
The syntax is a bit verbose, fine with me if you prefer to keep it as a nested if